dbcon/joblist/tuple-bps.cpp

```diff
@@ -2265,11 +2269,14 @@ void TupleBPS::receiveMultiPrimitiveMessages(uint32_t threadID)
                                         // the filter eliminated all matches, need to join with the NULL row
                                         if (matchCount == 0 && tjoiners[j]->largeOuterJoin())
                                         {
-                                            newJoinerOutput.push_back(Row::Pointer(smallNullMemory[j].get()));
+//                                            newJoinerOutput.push_back(Row::Pointer(smallNullMemory[j].get()));
+                                            newJoinerOutput.push_back(smallNulls[j].getPointer());
                                             matchCount = 1;
                                         }
 
                                         joinerOutput[j].swap(newJoinerOutput);
+                                        if (joinFERow.usesStringTable() && joinFERowData.getStringTableMemUsage() > 50 * (1 << 20))
+                                            joinFERowData.clearStringStore();
```

Comment:
We should move this threshold value (50 * (1 << 20)) into a common location (maybe in rowgroup.h) as a const int or a macro, so it can be easily updated later.

Use of UNLIKELY around the if predicate might be a suitable optimization?

We are missing a line break before the if condition.